### ISSUE TITLE
Unit tests fail for class LeftAlignAndTrimVariantsUnitTest on master 

### ISSUE THREAD

**Issue by rick-heig** [broadinstitute/gatk#5511]
Hello.
I am not sure if this is a bug or simply a tool version problem but when running the tests with `./gradlew test` I get 18 failures with the `org.broadinstitute.hellbender.tools.walkers.variantutils.LeftAlignAndTrimVariantsUnitTest` class.

I wanted to make sure GATK was working right when compiled it from source to have a working base a I intend to try out some experimental modifications to the code.

### Affected tool(s) or class(es)
`org.broadinstitute.hellbender.tools.walkers.variantutils.LeftAlignAndTrimVariantsUnitTest`

### Affected version(s)
- [ ] Latest public release version [version?]
- [x] Latest master branch as of [11.12.18]

### Description
The following commands were used to build and test GATK
```
git pull
./gradlew clean
./gradlew bundle
./gradlew test
```
The tests resulted in 18 failed as can be seen in the attached file. 
[Test results - Class org.broadinstitute.hellbender.tools.walkers.variantutils.LeftAlignAndTrimVariantsUnitTest.html.zip](https://github.com/broadinstitute/gatk/files/2667609/Test.results.-.Class.org.broadinstitute.hellbender.tools.walkers.variantutils.LeftAlignAndTrimVariantsUnitTest.html.zip)

If this is normal (expected) when building from the last commit on master you can close this issue. For experimental development should I use the most recent release or can I work from the most recent commit on master ?

#### Steps to reproduce
see commands above description

The problem could be from a library or java version maybe. I run a Ubuntu 16.04 LTS desktop.
```
uname -a
Linux A13PC04 4.4.0-140-generic #166-Ubuntu SMP Wed Nov 14 20:09:47 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

javac -version
javac 1.8.0_102

java -version
java version "1.8.0_102"
Java(TM) SE Runtime Environment (build 1.8.0_102-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.102-b14, mixed mode)
```

#### Expected behavior
I was expecting the tests to pass.

#### Actual behavior
18 tests failed.

**Comment by cmnbroad** [comment]
@rick-heig I wouldn't expect to see those failures, so I think something is wrong. Are these the only tests in the entire suite that fail ? We pretty regularly run these tests on Ubuntu 16.04. Also please verify that you did install git-lfs and fetched the large files per the instructions in the README.md file ?

**Comment by rick-heig** [comment]
I get : `500032 tests completed, 20 failed, 1 skipped`

The 2 extra failed tests are spark related (I don't have a spark setup so I assume this is normal)
`ExampleAssemblyRegionWalkerSparkIntegrationTest. testExampleAssemblyRegionWalkerNonStrict`
`ExampleAssemblyRegionWalkerSparkIntegrationTest. testExampleAssemblyRegionWalkerStrict`
The skipped test is `testLikelihoodsFromHaplotypes[2](null, false)` in the `VectorPairHMMUnitTest`

I did install git-lfs an downloaded all the required files per the instructions, the files are up to date.

**Comment by rick-heig** [comment]
Hello.
If I launch the tests by themselves with the following command they all pass
```
$ ./gradlew test --tests org.broadinstitute.hellbender.tools.walkers.variantutils.LeftAlignAndTrimVariantsUnitTest
...
Results: SUCCESS (3611 tests, 3611 successes, 0 failures, 0 skipped)

BUILD SUCCESSFUL

Total time: 22.445 secs
```

But when launching all the tests altogether I get
```
$ ./gradlew test
...
Results: FAILURE (500032 tests, 500011 successes, 20 failures, 1 skipped)

500032 tests completed, 20 failed, 1 skipped
...
```

**Comment by cmnbroad** [comment]
@rick-heig Ah - thats super helpful - I think I see the problem. Is is possible for you try running the tests with [this branch](https://github.com/broadinstitute/gatk/tree/cn_left_align) ?

**Comment by cmnbroad** [comment]
@rick-heig Oh wait - I pushed that commit too fast. I'll let you know when there is a branch to try.

**Comment by cmnbroad** [comment]
@rick-heig Ok - if you can give [this branch](https://github.com/broadinstitute/gatk/tree/cn_left_align) a try.

**Comment by rick-heig** [comment]
Ok, this seems to solve it. I ran :
```
$ git checkout cn_left_align 
$ ./gradlew clean
$ ./gradlew bundle
$ ./gradlew test
...
Results: FAILURE (500060 tests, 500057 successes, 2 failures, 1 skipped)

500060 tests completed, 2 failed, 1 skipped
```

And got all the tests to succeed (except the two spark related ones as before).

**Comment by cmnbroad** [comment]
Ok, thanks for validating that. I'll make a PR for this change (the way those unit tests are written is a little sketchy so I may fix that at the same time).

I forgot about the spark failures - can you post the log output for those failures as well ?

Also, to answer your original question, you should generally always be able to work directly from head of master and all tests should pass, though sometimes things like this can slip through.





Also, to answer your original question

**Comment by rick-heig** [comment]
Hello.
Thank you for your answer.

Attached are the logs of the Spark failures.
[org.broadinstitute.hellbender.tools.examples.ExampleAssemblyRegionWalkerSparkIntegrationTest.html.zip](https://github.com/broadinstitute/gatk/files/2685031/org.broadinstitute.hellbender.tools.examples.ExampleAssemblyRegionWalkerSparkIntegrationTest.html.zip)

**Comment by cmnbroad** [comment]
@tomwhite @jamesemery Any insight as to why these `ExampleAssemblyRegionWalkerSpark` tests might be failing ?

**Comment by tomwhite** [comment]
It's not immediately obvious why the Spark tests are failing. It could be something to do with the Spark context which is shared between tests, and which may be picking up some state that is not cleared from one test to the next. The tests are passing on Travis, which also runs all of them.

Do they fail if you run them individually?

**Comment by rick-heig** [comment]
Sorry for the delay, when running the tests individually I get : 
```
./gradlew test --tests org.broadinstitute.hellbender.tools.examples.ExampleAssemblyRegionWalkerSparkIntegrationTest
...
esults: SUCCESS (2 tests, 2 successes, 0 failures, 0 skipped)

BUILD SUCCESSFUL
```
So they seem to work when launched individually, like the other tests.
So it seem like there is a similar problem for these tests.

**Comment by tomwhite** [comment]
@rick-heig thanks for reporting back. It does look like it's a problem with Spark state then. I will see if I can reproduce the problem.

**Comment by tomwhite** [comment]
When I run the whole test suite with `./gradlew test` ExampleAssemblyRegionWalkerSparkIntegrationTest passes (both its tests). So I'm not sure how to reproduce this, which makes it a bit tricky to fix.

BTW I'm using a later version of Java, not sure if this is relevant though.

```
java version "1.8.0_171"
Java(TM) SE Runtime Environment (build 1.8.0_171-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.171-b11, mixed mode)
```

**Comment by rick-heig** [comment]
It doesn't seem related to the version of java. I updated my version of java to
```
javac -version
javac 1.8.0_191

java -version
java version "1.8.0_191"
Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)
```

The issue remains the same, I pulled the latest commit on master and ran
```
./gradlew clean
./gradlew bundle
./gradlew test
...
Results: FAILURE (500075 tests, 500072 successes, 2 failures, 1 skipped)

500075 tests completed, 2 failed, 1 skipped
```
The behavior is the same, the exact two same tests fail.

**Comment by tomwhite** [comment]
@rick-heig thanks for trying again. I ran the tests on Ubuntu 18.04 (the previous attempts were on a Mac), and the Spark tests didn't fail for me there either...

**Comment by davidbenjamin** [comment]
Closed by #5519.
